ios/browser/api/url/url_spoof_checker.h

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_URL_URL_SPOOF_CHECKER_H_
+#define BRAVE_IOS_BROWSER_API_URL_URL_SPOOF_CHECKER_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// The |SkeletonType| and |TopDomainEntry| are mirrored in trie_entry.h. These
+/// are used to insert and read nodes from the Trie.
+/// The type of skeleton in the trie node.
+NS_SWIFT_NAME(URLSpoofChecker.SkeletonType)
+typedef NS_ENUM(NSUInteger, BraveSpoofCheckerSkeletonType) {
+  BraveSpoofCheckerSkeletonTypeFull,
+  BraveSpoofCheckerSkeletonTypeSeparatorsRemoved
+};
+
+NS_SWIFT_NAME(URLSpoofChecker.LookalikeURLMatchType)
+typedef NS_ENUM(NSUInteger, BraveSpoofCheckerLookalikeURLMatchType) {
+  BraveSpoofCheckerLookalikeURLMatchTypeNone = 0,
+  /// DEPRECATED: Use kSkeletonMatchTop500 or kSkeletonMatchTop5k.
+  /// kTopSite = 1,
+  BraveSpoofCheckerLookalikeURLMatchTypeSiteEngagement = 2,
+  BraveSpoofCheckerLookalikeURLMatchTypeEditDistance = 3,
+  BraveSpoofCheckerLookalikeURLMatchTypeEditDistanceSiteEngagement = 4,
+  BraveSpoofCheckerLookalikeURLMatchTypeTargetEmbedding = 5,
+  BraveSpoofCheckerLookalikeURLMatchTypeSkeletonMatchTop500 = 6,
+  BraveSpoofCheckerLookalikeURLMatchTypeSkeletonMatchTop5k = 7,
+  BraveSpoofCheckerLookalikeURLMatchTypeTargetEmbeddingForSafetyTips = 8,
+
+  /// The domain name failed IDN spoof checks but didn't match a safe hostname.
+  /// As a result, there is no URL to suggest to the user in the form of "Did
+  /// you mean <url>?".
+  BraveSpoofCheckerLookalikeURLMatchTypeFailedSpoofChecks = 9,
+
+  BraveSpoofCheckerLookalikeURLMatchTypeCharacterSwapSiteEngagement = 10,
+  BraveSpoofCheckerLookalikeURLMatchTypeCharacterSwapTop500 = 11
+};
+
+OBJC_EXPORT
+NS_SWIFT_NAME(URLSpoofChecker.TopDomainEntry)
+@interface URLSpoofCheckerTopDomainEntry : NSObject
+/// The domain name.
+@property(nonatomic, readonly) NSString* domain;
+/// True if the domain is in the top 500.
+@property(nonatomic, readonly) bool isTop500;
+/// Type of the skeleton stored in the trie node.
+@property(nonatomic, readonly) BraveSpoofCheckerSkeletonType skeletonType;
+@end
+
+OBJC_EXPORT
+NS_SWIFT_NAME(URLSpoofChecker.Result)
+@interface BraveURLSpoofCheckerResult : NSObject
+@property(nonatomic, readonly) bool isLookalikeURL;
```

Comment:
Isn't the match type not being none enough information to determine this? Seems like this property is not useful